e2e/tests/node.go

```diff
@@ -263,3 +261,24 @@ func formatURLString(serviceAlias, rpcUrl, path string) string {
 	}
 	return fmt.Sprintf("http://%s.%s/v1/%s", serviceAlias, rpcUrl, path)
 }
+
+// getVirtualHostFromHost returns the given rpcUrl host stripped of the port
+// and with the service included as a subdomain to be used in the "Host" request header.
+// This is needed by the current DevNet infrastructure that does not support arbitrary
+// subdomains which the PATH Gateway needs to identify the requested service.
+func getVirtualHostFromHost(rpcUrl *url.URL, service string) string {
+	// Store the virtual host we want to send in the Host header
+	virtualHost := rpcUrl.Host
+
+	if len(service) > 0 {
```

Comment:
Can you #PUC when this would be zero?

Reply:
This should never be zero, and it's not the responsibility of this function to check it.

I'll move the check to where it's relevant.